### PR TITLE
feat(benchmarking): working benchmarking extrinsic indexing

### LINKED ISSUE #107: Index RPC calls and add expected weight consumption
Indexing extrinsics and adding expected weight computation

### PR DESCRIPTION
**Pull request by Supremesource**
This PR:
- adds the ability to run benchmarks for all mainnet extrinsic calls
- it completely deletes the old benchmarking replacing it with new logic.
- indexes extrinsic calls 
- adding expected weight consumption to every extrinsic

Renaming: 
- `autogen_weights` -> `weights`
- `weights` -> `set_weights`
Closes #107

### CODE REVIEW

File: node/src/service.rs [hunk 1]
```diff
@@ -17,6 +17,26 @@ pub(crate) type FullClient =
 type FullBackend = sc_service::TFullBackend<Block>;
 type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
 
+#[allow(dead_code)]
```
Comment (devwckd) [inline node/src/service.rs:20]: What is this used for?

Non blocking

Comment (Supremesource) [reply node/src/service.rs:20]: Oh, I will delete this

**Review by devwckd — APPROVED** [review]
LGTM

File: node/src/main.rs [hunk 2]
```diff
@@ -5,6 +5,8 @@
 mod chain_spec;
 #[macro_use]
 mod service;
+
+// #[cfg(feature = "runtime-benchmarks")]
```
Comment (saiintbrisson) [inline node/src/main.rs:9]: is this correct?

File: Makefile [hunk 3]
```diff
@@ -23,3 +22,7 @@ RUST_LOG ?= info,pallet_subspace::migrations=debug
 try-runtime-upgrade:
 	cargo build --release --features try-runtime
 	RUST_BACKTRACE=1; RUST_LOG="${RUST_LOG}"; try-runtime --runtime target/release/wbuild/node-subspace-runtime/node_subspace_runtime.compact.compressed.wasm on-runtime-upgrade live --uri wss://commune-api-node-0.communeai.net:443
+run-benchmarking:
+	cargo build -r --features runtime-benchmarks
+	./target/release/node-subspace build-spec --disable-default-bootnode --chain local > specs/benchmarks.json
+	./target/release/node-subspace benchmark pallet --chain specs/benchmarks.json --pallet pallet_subspace --extrinsic "*" --steps 50 --repeat 20 --output pallets/subspace/src/weights.rs --template=./.maintain/frame-weight-template.hbs
```
Comment (saiintbrisson) [inline Makefile:28]: it's missing a new line here

**Review by saiintbrisson — CHANGES_REQUESTED** [review]

File: node/src/main.rs (hunk 2 above)
Comment (Supremesource) [reply node/src/main.rs:9]: yes

**Pull request merged**